pkg/e2e/operators/cloudingress/certificate.go

```diff
@@ -0,0 +1,71 @@
+package cloudingress
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/osde2e/pkg/common/constants"
+	"github.com/openshift/osde2e/pkg/common/helper"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// tests
+var _ = ginkgo.Describe(constants.SuiteInforming+TestPrefix, func() {
+	h := helper.New()
+
+	ginkgo.Context("publishingstrategy-certificate", func() {
+		ginkgo.It("IngressController should be patched when update Certificate", func() {
+			log.Print("Gonna update Certificate")
+			updateCertificate(h, "foo-bar")
+			time.Sleep(time.Duration(60) * time.Second)
+			log.Print("Updated the certificate in Publishingstrategy")
+			ingress, _ := getingressController(h, "default")
+			temp := (int64(1))
+			Expect(string(ingress.Spec.DefaultCertificate.Name)).To(Equal("foo-bar"))
+			Expect(ingress.Generation == temp).To(Equal(false))
+			Expect(ingress.Annotations["Owner"]).To(Equal("cloud-ingress-operator"))
+			log.Print("Ok Done with the test")
+		})
+		ginkgo.It("IngressController should be patched when return the original Certificate", func() {
+			log.Print("Gonna return the origin certificate")
+			updateCertificate(h, "linnguye-primary-cert-bundle-secret")
```

Comment:
you might have to get the original certname before changing it in the previous test. 
Another strategy could be to append or prepend to the existing name. which might make it easier to revert back to original name.